src/storage/observable/fromTask.ts

```diff
@@ -0,0 +1,14 @@
+import { storage } from 'firebase/app';
+import 'firebase/storage';
+import { Observable } from 'rxjs/Observable';
+
+export function fromTask(task: storage.UploadTask) {
+  return new Observable<storage.UploadTaskSnapshot | undefined>(subscriber => {
+    task.on('state_changed', 
```

Comment:
FYI I don't know if it was fixed since but I found that `state_changed` is not updated right before `complete()` in the JS SDK, say if you had a percentage uploader it would never hit 100%.

In my prior WIP branch I got this working nicely via:

    task.on('state_changed', subscriber.next);
    task.then(snapshot => {
        subscriber.next(snapshot);
        subscriber.complete();
    }).catch(e => subscriber.error);

Reply:
From my docs before, make sure these all fire as expected:

```
this.state = task.map(s => s.state);
this.uploading = state.map(s => s === firebase.storage.TaskState.RUNNING);
this.success = state.map(s => s === firebase.storage.TaskState.SUCCESS);
this.percentage = task.map(s => s.bytesTransferred / s.totalBytes * 100);
```

That WIP was here FYI https://github.com/angular/angularfire2/compare/master...jamesdaniels:storage

Reply:
Pinging @sphippen!

Reply:
Running and percentage are good. I can say that `this.success = state.map(s => s === firebase.storage.TaskState.SUCCESS);` does not fire. The bug may still persist. Let's get Spencer's opinion on how to handle.

Reply:
Hmm, is the problem just that you don't get an event inside the 'state_change' stream that indicates 100% progress? Based on my understanding of the code, that shouldn't be happening. I'll test it out.

Reply:
I tested some code that looks like this:

```
var task = ref.put(getFiles()[0]); task.on('state_changed', (snap) => { console.log(snap.bytesTransferred + ' / ' + snap.totalBytes); }, () => { console.log('error'); }, () => { console.log('done'); });
```

and the snapshot callback was called with bytesTransferred == totalBytes before the "done" callback was called.

Is that consistent with what you are seeing? I don't understand the context here very much, so it'd help if you could make a MCVE of the thing you consider a bug.

Reply:
@sphippen Your code sample works, the problem we are having is that the progress callback never fires for the enum `firebase.storage.TaskState.SUCCESS`.

```ts
var task = ref.put(getFiles()[0]); 
task.on('state_changed', (snap) => { 
  if(snap.state === firebase.storage.TaskState.SUCCESS) {
     console.log('SUCCESS!');
  } 
}, () => { console.log('error'); }, () => { console.log('done'); });
```